### ISSUE TITLE
Snackbar

### ISSUE THREAD

**Issue by spiegelp** [ButchersBoy/MaterialDesignInXamlToolkit#390]
Are there any plans or active development for a Snackbar (https://www.google.com/design/spec/components/snackbars-toasts.html) and a Stepper (https://www.google.com/design/spec/components/steppers.html)? Otherwise I am going to implement them the next weekends.

**Comment by ButchersBoy** [comment]
Yeah, trying to work through all of the guidelines, I'll  probably hit snack bars first.

**Comment by spiegelp** [comment]
I start to implement a Snackback these days.

**Comment by spiegelp** [comment]
Is there any schedule for version 2.1.0? It would be nice to have a rough time frame for the next implementation steps. 

It's a little bit tricky to implement all of the details of the Stepper specs. However I already have a partially working vertical Stepper on my development branch. I'll continue it the next days.

**Comment by ButchersBoy** [comment]
I will start looking at this again shortly; I have some personal work which will utilise it so should go into 2.2

**Comment by ButchersBoy** [comment]
(renamed to Snackbar to account for that's what's going into 2.2)

**Comment by ButchersBoy** [comment]
Final proposed API now on snackbar branch, in this PR: https://github.com/ButchersBoy/MaterialDesignInXamlToolkit/pull/524

And also detailed in wiki: https://github.com/ButchersBoy/MaterialDesignInXamlToolkit/wiki/Snackbar

**Comment by spiegelp** [comment]
The current implementation works well. However I did not find a detail in the behaviour yet: Does new content interrupt an already visible Snackbar? I mean that a visible Snackbar hides immediatelly so the new content can be shown. At the moment I think that this needs another implementation of the queue interface.

**Comment by ButchersBoy** [comment]
If you use the queue, then it is just that; queued.  If you affect the .Content directly then it act as as any other content control.

**Comment by ButchersBoy** [comment]
@spiegelp yes...regarding the interruptions; I agree.  I was playing swiping through some Google Inbox messages the other day and you get the immediate snackbar.

I'm internally debating the API.  Either replacing ``` bool neverConsiderToBeDuplicate ``` with some sort of Enum, or add a whole new  ``` Now(. . .) ``` type of method (I've never been comfortable with other pump's Send/Post semantics; too confusing).

Any thoughts?

**Comment by spiegelp** [comment]
I was thinking of a new class inheriting from `ISnackbarMessageQueue` (something like `SnackbarSwapper`). `Enqueue()` will not add the content to a queue. It will replace a visible content directly (like the experimental and deleted `SnackbarHost` did). Although the method name does not describe the behaviour. That is the easy "let inherit and do it differently" way.

In my current imagination I am not happy with `Now()`. The behaviour of `Now()` will be in conflict with `Enqueue()` in `SnackbarMessageQueue`. Certain parts of the controller logic contradict inside the controller. I am afraid of the neccessary state management to prevent side effects. Usually a user does exactly what he is not supposed to do. So I would consider this approach as last fallback. 

Are you thinking of the `Enum` as some kind of switch for different behaviour inside one class (enqueue, replace, ...)? That might be an interesting approach. `ISnackbarMessageQueue` could change to an `ISnackbarController` with a method `Add()`. The method will act according to `enum SnackbarMode` or maybe delegates the work to specific internal classes. 

These are some thoughts on possible approaches. Sometimes you only see resulting problems of an idea if you write code. A short anecdote: This year, a two week task took me three months to implement. Our customers did not consider the side effects on several related processes that we found during coding. They were surprised of hearing about them and they had a hard time to decide how to move on. So I cannot guarantee that my current opinion will be a great solution before trying it. 

Maybe I confirmed your preference or I did the opposite.
